subprojects/gradle/module-dependencies-graph/src/main/kotlin/com/avito/module/internal/dependencies/DependenciesGraphBuilder.kt

```diff
@@ -13,50 +13,24 @@ internal class DependenciesGraphBuilder(
 ) {
 
     init {
-        require(root == root.rootProject)
+        require(root == root.rootProject) {
+            "Project $root must be the root"
+        }
     }
 
     private val logger = loggerFactory.create<DependenciesGraphBuilder>()
-    private val rootProjects = mutableListOf<Project>()
+
     private val visited = mutableMapOf<Project, ModuleProjectDependenciesNode>()
 
-    fun buildDependenciesGraph(): Set<ModuleProjectDependenciesNode> {
+    fun buildDependenciesGraph(): List<ModuleProjectDependenciesNode> {
```

Comment:
Now, this graph doesn't know about Android apps and produces only a model of a project.